### PR TITLE
Implement specification pattern

### PR DESCRIPTION
**Pull request by BlackkTurtle**
##Summary
- Implemented specification pattern in BaseRepository
- Added RelatedTerm GetById and GetAll endpoints in RelatedTermController
- RelatedTerm Handlers were updated with Specifications
- RelatedTerm Tests were updated with Specifications
- Tested the RelatedTermController and SpecificationEvaluator

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -346,3 +346,4 @@ healthchecksdb
 BlobStorageFolder/
 BlobStorageTestFolder/
 /Streetcode/sqlserver.env
+/houses.zip
```
Comment (vladtymo) [inline .gitignore:349]: It's might be better to make a separate task for this fix

File: Streetcode/Streetcode.XUnitTest/Streetcode.XUnitTest.csproj [hunk 2]
```diff
@@ -40,5 +40,6 @@
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\Streetcode.BLL\Streetcode.BLL.csproj" />
+    <ProjectReference Include="..\Streetcode.WebApi\Streetcode.WebApi.csproj" />
```
Comment (vladtymo) [inline Streetcode/Streetcode.XUnitTest/Streetcode.XUnitTest.csproj:43]: What's the reason to include Web Api here?

**Comment by vladtymo** [comment]
All code related to the Specification looks good

File: Streetcode/Streetcode.XUnitTest/Streetcode.XUnitTest.csproj (hunk 2 above)
Comment (BlackkTurtle) [reply Streetcode/Streetcode.XUnitTest/Streetcode.XUnitTest.csproj:43]: To test the RelatedTermController

**Review by max-khakimzhanov — APPROVED** [review]

**Review by synapsetron — APPROVED** [review]

**Pull request merged**